crypto/sha/keccak1600.c

```diff
@@ -11,8 +11,67 @@
 #include <string.h>
 #include <assert.h>
 
-#define ROL64(a, offset) ((offset) ? (((a) << offset) | ((a) >> (64-offset))) \
-                                   : a)
+#define ROL32(a, offset) (((a) << (offset)) | ((a) >> ((32 - (offset)) & 31)))
+
+static uint64_t ROL64(uint64_t val, int offset)
+{
+    if (offset == 0) {
+        return val;
+    } else if (sizeof(void *) == 8) {
```

Comment:
Wasn't this discussed? Wasn't it like you need extra -Wflag and "solution" was extra parenthesis all over?